core/classifiers/CodeClassifier/CodeClassifier.py

```diff
@@ -0,0 +1,710 @@
+# coding: utf-8
+#
+# Copyright 2017 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base class for classification algorithms."""
+
+import collections
+import keyword
+import math
+import StringIO
+import token
+import tokenize
+
+from core.classifiers import base
+from core.classifiers.CodeClassifier import winnowing
+
+import numpy as np
+from sklearn.feature_extraction import text as sklearn_text
+from sklearn import model_selection
+from sklearn import svm
+
+# The string with which all the variable and method names need to be replaced.
+VAR_TOKEN = 'V'
```

Comment:
The usage of "token" seems inconsistent throughout. Sometimes I see "token" and sometimes I see "token name" referring to the same sort of thing. Let's pick one and standardize accordingly. For example, if we want to go with "token_name" then maybe call this VAR_TOKEN_NAME and use "token" to mean the ID + name.